app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# app/controllers/sessions_controller.rb
+class SessionsController < ApplicationController
+  def omniauth
+    user = SessionsFacade.new.authorize_user(request.env["omniauth.auth"])
+    session[:user_id] = user.name
```

Comment:
Brilliant idea to save the user name so we don't need to ask BE for that data at any other point throughout a session!